### ISSUE TITLE
I can't compile a project with    Compiling polars-io v0.38.2 error[E0658]: use of unstable library feature 'arc_unwrap_or_clone'  

### ISSUE THREAD

**Issue by xsirio** [pola-rs/polars#14988]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust
//it seems lazy package problem 
polars = {version="0.38.2",features =["lazy","parquet","dtype-datetime"]}
```

### Log output

```shell
error[E0658]: use of unstable library feature 'arc_unwrap_or_clone'
  --> /Users/nicsir/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-lazy-0.38.2/src/physical_plan/executors/scan/csv.rs:21:18
   |
21 |             .map(Arc::unwrap_or_clone);
   |                  ^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #93610 <https://github.com/rust-lang/rust/issues/93610> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `polars-lazy` (lib) due to previous error
```

### Issue description

trying to compiling a small program with lazy features. Compile error.

### Expected behavior

successful compilation of polars-lazy lib... 

### Installed versions

<details>

polars = {version="0.38.2",features =["lazy","parquet","dtype-datetime"]}

</details>

**Comment by evbo** [comment]
Before the `0.38.2` release, polars `0.38.0` worked great but now every version of polars `> 0.37.0` fails due to this.

**Comment by eitsupi** [comment]
Have you checked the issue link?

> note: see issue #93610 <https://github.com/rust-lang/rust/issues/93610> for more information

The issue is fixed in Rust 1.76.0.
In the other words, MSRV is now 1.76.0. You have to update rustc.

Related issue #12112

**Comment by xsirio** [comment]
Thanks!
